presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyCountOverInnerAggregateQuery.java

```diff
@@ -60,8 +63,7 @@
         if (!(child instanceof AggregationNode)) {
             return Optional.empty();
         }
-        AggregationNode childAggregationNode = (AggregationNode) child;
-        if (childAggregationNode.getAssignments().isEmpty() && childAggregationNode.hasEmptyGroupingSet() && !childAggregationNode.hasNonEmptyGroupingSet()) {
+        if (ScalarQueryUtil.isScalar(child, lookup)) {
```

Comment:
Once you used this method, your rule become more generic as it can handle source like EnsureSingleRow or Values with single row, so you need to change the rule name.